pinot-common/src/main/java/org/apache/pinot/pql/parsers/Pql2Compiler.java

```diff
@@ -152,6 +157,69 @@ public BrokerRequest compileToBrokerRequest(String expression)
     }
   }
 
+  /**
+   * Currently we don't support DISTINCT queries that have ORDER BY
+   * Until support for ORDER BY with DISTINCT is implemented,
+   * throw compilation error
+   *
+   * We walk the AST to detect if the AST has a FunctionCallAstNode for
+   * DISTINCT function along with OrderByAstNode.
+   *
+   * The following indicates how AST generally looks like for DISTINCT
+   * queries. Under the root node (SelectAstNode), we don't need to check
+   * for all kinds of children (like WhereAstNode, GroupByAstNode,
+   * StarColumnListAstNode).
+   *
+   * Only check for 2 children: OutputColumnListAstNode (and ensure output column
+   * inside this list is for DISTINCT function) and OrderByAstNode
+   *
+   * SelectAstNode{_tableName='foo', _resourceName='foo', _recordLimit=-1, _offset=-1}
+   *     OutputColumnListAstNode
+   *       OutputColumnAstNode
+   *         FunctionCallAstNode{_name='DISTINCT'}
+   *           IdentifierAstNode{_name='col1'}
+   *           IdentifierAstNode{_name='col2'}
+   *     OrderByAstNode
+   *       OrderByExpressionAstNode{_column='col1', _ordering='asc'}
+   *         IdentifierAstNode{_name='col1'}
+   *       OrderByExpressionAstNode{_column='col2', _ordering='asc'}
+   *         IdentifierAstNode{_name='col2'}
+   * @param root root node of AST
+   */
+  private void checkForOrderByWithDistinct(AstNode root) {
+    if (root instanceof SelectAstNode) {
```

Comment:
This check is redundant

Reply:
Done. Moved validation to BaseBrokerRequestHandler